packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -284,12 +284,84 @@ class NestedScrollView extends StatefulWidget {
   }
 
   @override
-  _NestedScrollViewState createState() => _NestedScrollViewState();
+  NestedScrollViewState createState() => NestedScrollViewState();
 }
 
-class _NestedScrollViewState extends State<NestedScrollView> {
+/// The [State] for a [NestedScrollView].
+///
+/// It allows you to obtain the [ScrollController]s of the children of your [NestedScrollView].
+///
+/// You can access the inner scroll controller via [innerController] and the outer
+/// controller with [outerController], however, notice that the outer scroll position
+/// can also be obtained using [NestedScrollView.controller].
+///
+/// If you want to access the inner scroll controller of your nested scroll view,
+/// you can get its [NestedScrollViewState] by supplying a `GlobalKey<NestedScrollViewState>`
+/// to your [NestedScrollView] widget (the `key` parameter).
+///
+/// {@tool sample}
+///
+/// You would access the [NestedScrollViewState] using a [GlobalKey].
+/// Using the following setup, you can access the inner scroll controller
+/// using `globalKey.currentState.innerController`.
+///
+/// ```dart
+/// class ExampleWidget extends StatelessWidget {
+///   final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return NestedScrollView(
+///       key: globalKey,
+///       headerSliverBuilder: (context, _) => const [SliverAppBar()],
+///       body: const CustomScrollView(
+///         // ...
+///       ),
+///     );
+///   }
+///
+///   ScrollController get innerController {
+///     return globalKey.currentState.innerController;
+///   }
+/// }
+/// ```
+///
+/// {@end-tool}
+///
+/// Alternatively, you could also visit child elements of your parent widget to obtain
```

Comment:
https://api.flutter.dev/flutter/widgets/Element/visitChildren.html

It might be the better option in some cases, but most people do not know how to use it I suppose.

Reply:
Let's just remove this paragraph. The other way of obtaining the controller is better.